shoop/xtheme/views/extra.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+import six
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.core.signals import setting_changed
+from django.http.response import HttpResponseNotFound
+
+from shoop.utils.importing import load
+from shoop.xtheme.theme import get_current_theme
+
+_VIEW_CACHE = {}
+
+
+def clear_view_cache(**kwargs):
+    _VIEW_CACHE.clear()
+
+
+setting_changed.connect(clear_view_cache, dispatch_uid="shoop.xtheme.views.extra.clear_view_cache")
+
+
+def _get_view_by_name(theme, view_name):
+    view = getattr(theme, "views", {}).get(view_name)
+    if isinstance(view, six.string_types):  # Handle loading views from specs
+        try:
+            view = load(view)
+        except ImproperlyConfigured:
+            if settings.DEBUG:
+                raise
+            return None
```

Comment:
The `get_views` is an advanced feature anyway; you'd have to write the view too and that means, you'd know something about Python.

No matter how this is implemented, the only way to make it accessible for average theme designer, is to have good documentation that tells how to create such custom views. It's just copy-paste from the documentation being it a dict of strings or a method returning a dict of objects with couple import statements.